### PR TITLE
vim-patch:partial:f10911e5db16

### PR DESCRIPTION
**Pull request by clason**
Update runtime files
https://github.com/vim/vim/commit/f10911e5db16f1fe6ab519c5d091ad0c1df0d063

skip:
* doc/options.txt (needs 8.2.4193)
* doc/change.txt (needs https://github.com/neovim/neovim/pull/17228)
* doc/index.txt (needs https://github.com/neovim/neovim/pull/17228)
* doc/eval.txt (error code shuffling)

### CODE REVIEW

File: runtime/doc/builtin.txt [hunk 1]
```diff
@@ -1723,6 +1723,22 @@ exists({expr})	The result is a Number, which is |TRUE| if {expr} is
 		For checking if a file exists use |filereadable()|.
 
 		The {expr} argument is a string, which contains one of these:
+			varname		internal variable (see
+			dict.key	|internal-variables|).  Also works
+			list[i]		for |curly-braces-names|, |Dictionary|
+			import.Func	entries, |List| items, imported
+					items,etc.
+					Does not work for local variables in a
+					compiled `:def` function.
```
Comment (seandewar) [inline runtime/doc/builtin.txt:1732]: `import.Func` and `:def` stuff snuck its way in

**Review by seandewar — COMMENTED** [review]
Missing
- https://github.com/vim/vim/commit/f10911e5db16f1fe6ab519c5d091ad0c1df0d063#diff-442131dbc53d601ff38edcc95c73c09272934f92f7cf6601738cc6ceaf8a1e48R7789 and the changes below.
- https://github.com/vim/vim/commit/f10911e5db16f1fe6ab519c5d091ad0c1df0d063#diff-6a5ba7723775463d3efe8549c14e5d64425287328acfe1dbc0539c47a7286df1R1185
- https://github.com/vim/vim/commit/f10911e5db16f1fe6ab519c5d091ad0c1df0d063#diff-323629ad524194bb982818dfbde19891f648730790774aa4955261178ffd73f0R1005-R1006 (could also apply the change to `O` if you'd like (excluding changes to `P`))

File: runtime/autoload/freebasic.vim [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  endif
+
+  " override with #lang directive or metacommand
+
+  let skip = "has('syntax_items') && synIDattr(synID(line('.'), col('.'), 1), 'name') =~ 'Comment$'"
+  let pat = '\c^\s*\%(#\s*lang\s\+\|''\s*$lang\s*:\s*\)"\([^"]*\)"'
+
+  let save_cursor = getcurpos()
+  call cursor(1, 1)
+  let lnum = search(pat, 'n', '', '', skip)
```
Comment (seandewar) [inline runtime/autoload/freebasic.vim:26]: Hmm, actually this needs https://github.com/vim/vim/commit/adc17a5f9d207fd1623fd923457a46efc9214777

**Review by seandewar — COMMENTED** [review]
Pedantic, but I guess you could consider adding https://github.com/vim/vim/commit/f10911e5db16f1fe6ab519c5d091ad0c1df0d063#diff-442131dbc53d601ff38edcc95c73c09272934f92f7cf6601738cc6ceaf8a1e48R1561-R1562 as it kinda applies to `v:true`/`v:false` too, but I feel like it's more-so related to Vim9 script where the `v:` is omitted (though we do have some mentions of just `true` and `false` in our Vim script docs too).

File: runtime/autoload/freebasic.vim (hunk 2 above)
Comment (clason) [reply runtime/autoload/freebasic.vim:26]: Ouch :/

Comment (seandewar) [reply runtime/autoload/freebasic.vim:26]: Very sneaky too! I'll see if it can be ported

Comment (clason) [reply runtime/autoload/freebasic.vim:26]: Or the `skip` be avoided...

Comment (seandewar) [reply runtime/autoload/freebasic.vim:26]: Maybe. At first glance seems to be used to ignore language directives that have been commented out.

Comment (seandewar) [reply runtime/autoload/freebasic.vim:26]: We could just ignore the skip for now.
Same is done if `syntax_items` aren't supported, so it's probably not the end of the world :P

Comment (clason) [reply runtime/autoload/freebasic.vim]: Makes sense; I left the line commented out with a mention of the missing patch so it can be revisited.

**Review by seandewar — APPROVED** [review]
LGTM. After a quick paranoia-induced re-review I'm pretty confident there's no more surprises. 😄

**Comment by clason** [comment]
These are getting trickier and trickier, aren't they...

**Pull request merged**